tests/serialization/test_kernel_serialization.py

```diff
@@ -2,12 +2,12 @@
 
```

Comment:
please add a simple iteration test (in test_iterations.py) that tests kernels and their combinations, akin to what been done in #225 for prios

Reply:
I have rebased this PR to sit on top of #225 and used your existing architecture. I however had to restrict the tests a bit due to numerical issues, I hope that this is still fine.

Reply:
perhaps I would possibly restrict the tests even more via pinning to one `nu`
this is jsut toa void creating a huge amount of tests for all settings, especially if more kernels are added at a later point, Id have chosen one setting per kernel type + scaled/notscaled

Reply:
Adjusted the test. For now, the `ScaleKernel` still tests all of the priors, but I can also easily remove that, just un-resolve this comment in case I should change this.